api/objects.proto

```diff
@@ -73,6 +73,20 @@ message Service {
 	// "Spec".
 	ServiceSpec previous_spec = 6;
 
+	// InterpolatedSpec is a version of Spec that has unset fields filled
+	// in with the default values. The client can use this field to see
+	// actual parameter values for all fields, unlike Spec, which is
+	// completely under the client's control, and therefore only has
+	// values for fields that have been set by the client.
+	//
+	// TODO(aaronl): Any object that follows the spec abstraction can
+	// benefit from having an InterpolatedSpec. Right now, this is most
+	// useful for services, because "service inspect" is frequently used,
+	// and it is useful for that to return detailed information about
+	// what values are actually being used to run the service. In the
+	// future, we should add add InterpolatedSpec to other objects as well.
+	ServiceSpec interpolated_spec = 7;
```

Comment:
I agree with this as far as the design.

Do you think the user should be able to view both the service spec they provided, and the full spec on inspect? As a user, maybe I don't want to always see the defaults.